database.cc

```diff
@@ -1407,7 +1407,7 @@ compare_atomic_cell_for_merge(atomic_cell_view left, atomic_cell_view right) {
     return 0;
 }
 
-future<std::tuple<lw_shared_ptr<query::result>, cache_temperature>>
+future<std::tuple<lw_shared_ptr<query::result>, cache_temperature, query::status>>
```

Comment:
Ok, after some more discussions about RPC, I understand now that the intention was that these functions should return *either* a result *or* a non-OK status. I think this needs to be clearly documented, it's really not obvious.

But @psarna I think I have an even better (maybe...) idea:
You can leave this function's signature unchanged, and translate a non-OK status into an exceptional future return. You need this weird "status" thing in the RPC function, but that is the *only* place where you need it! In all the layers you now added "status" too, you could have left the unchanged if the lowest later translates the non-OK status into an expection

Reply:
Adding `query::status` here was not done to propagate exceptions, it's actually just a convenience change to avoid adding another continuation to provide a `OK` status. With this function returning a status, the RPC handler looks like that (pseudocode):
```cpp
return query_result_local(stuff).handle_exception_type([] (overloaded_exception& e {
    return ready_future_with_overloaded_exception;
})
```
, while if I leave this function's declaration as it was before, I'd have to do:
```cpp
return query_result_local(stuff).then([] (std::tuple<query_result, cache_temperature>) {
    return make_ready_future(query_result, cache_temperature, status::OK);
}).handle_exception_type([] (overloaded_exception& e {
    return ready_future_with_overloaded_exception;
})
```

Still, I agree there is value in keeping the interfaces clean, so I'll try removing status from non-storage-proxy functions and see if it looks alright.